src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -189,19 +200,21 @@ public PolicyEntity call() throws Exception {
 
             @Override
             public List<String> call() throws Exception {
-                CriteriaBuilder cb = em.getCriteriaBuilder();
-                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
-                Root<PolicyEntity> r = q.from(PolicyEntity.class);
-                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)));
-                List<String> policy = new ArrayList<String>();
-                List<PolicyEntity> PolicyList = em.createQuery(q).getResultList();
-                if (PolicyList.isEmpty() == false) {
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("domain").get("Id"), domainid)));
+                List<String> policy = null;
+                List<PolicyEntity> PolicyList = em.createQuery(query).getResultList();
```

Comment:
can you just return the resultLisT?

Reply:
No, the resultlist is the list of policies and we need list of ID's to be returned.